backend/BalanceBuddy/src/main/java/com/example/balancebuddy/config/SecurityConfiguration.java

```diff
@@ -57,7 +58,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .authorizeHttpRequests((authorize) -> authorize
                         // Permit all requests to login and register endpoints
                         .requestMatchers("/api/auth/login", "/api/auth/register", "/api/auth/token").permitAll()
-                        .anyRequest().authenticated() // All other requests must be authenticated
+                        .requestMatchers("/api/user/change-password").authenticated()
+                        .anyRequest().authenticated()// All other requests must be authenticated

```

Comment:
You can leave just ".anyRequest().authenticated()". I think "/api/user/change-password" endpoint should be included in anyRequest() method.